staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -754,14 +754,35 @@ func shouldDelegateList(opts storage.ListOptions) bool {
 	consistentReadFromStorage := resourceVersion == "" && !(consistentListFromCacheEnabled && requestWatchProgressSupported)
 	// Watch cache doesn't support continuations, so serve them from etcd.
 	hasContinuation := len(pred.Continue) > 0
-	// Serve paginated requests about revision "0" from watch cache to avoid overwhelming etcd.
-	hasLimit := pred.Limit > 0 && resourceVersion != "0"
+	// Serve paginated requests about revision "0" and "" (when consistent read from cache is supported)
```

Comment:
Whenever you change this function, please also update the one in estimator - see the comment 10 lines above.

Reply:
good catch, thanks.

Reply:
do you know why we have two copies of the same function ?

Reply:
Discussed offline.